movie_search/media.py

```diff
@@ -13,32 +13,15 @@
     TVSeriesRecommendation,
     TVSeriesProductionCompany,
 )
-from .tmdb_api import TMDBApi
-
-tmdb_api_obj = TMDBApi()
-
-
-# standalone functions for general fetching of data
-def fetch_data_from_api(endpoint, **kwargs):
-    """A common method to fetch data from media API"""
-    return tmdb_api_obj.get_data_from_endpoint(endpoint, **kwargs)
-
-
-def fetch_api_data_by_query(endpoint, text_query, results_key):
-    """A common method to fetch data by query from media API"""
-    return tmdb_api_obj.get_data_by_query(endpoint, text_query, results_key)
-
-
-def lookup_id_in_data_by_query(data_dict, query):
-    """A common method to fetch media ID from data by query"""
-    return {item.lower(): idx for item, idx in data_dict.items()}.get(query)
 
 
 class MediaService(ABC):
+    def __init__(self, tmdb_api):
```

Comment:
This can be considered a form of dependency injection, but it would make most sense when you anticipate the need to provide different implementations of a dependency. 

For instance, in the case of the tmdb_api object, you might have different types of API clients for different environments (development, testing, production) or different services (for example, an alternative movie database API), that's probably not the case here right?

We can leave it like this though, no harm done.

Reply:
Right this was the screenshot of the thread that you couldn't access @bbelderbos 
![image](https://github.com/vnoelifant/movie-search-app/assets/43191157/95b1c492-3617-485b-a472-586abb2c6430)
![image](https://github.com/vnoelifant/movie-search-app/assets/43191157/6d7b0387-162f-44fc-ab4a-995f93a7f97e)